File: mobile/composeApp/src/commonMain/kotlin/com/example/ultra/core/domain/model/CartItem.kt
package com.example.ultra.core.domain.model

data class CartItem(
    val id: String,
    val productId: String,
    val variantId: String = "",
    val title: String = "",
    val product: Product? = null,
    val quantity: Int = 1,
    val unitPrice: Double = 0.0,
    val total: Double = 0.0,
    val currency: String = "EUR",
    val imageUrl: String? = null
) {
    val subtotal: Double get() = unitPrice * quantity

    fun toProduct(): Product {
        return product ?: Product(
            id = productId,
            vendorId = "",
            name = title,
            description = "",
            price = unitPrice,
            currency = currency,
            imageUrl = imageUrl
        )
    }
}

File: backend-dotnet/src/AlturaNova.Domain/Entities/CartItem.cs
namespace AlturaNova.Domain.Entities;

/// <summary>A line item in a cart, referencing a specific product variant.</summary>
public class CartItem
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid CartId { get; set; }
    public Cart Cart { get; set; } = null!;

    public Guid ProductId { get; set; }
    public Product? Product { get; set; }

    public Guid VariantId { get; set; }
    public ProductVariant? Variant { get; set; }

    public string Title { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public string Currency { get; set; } = "EUR";
    public string? ThumbnailUrl { get; set; }

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public decimal LineTotal => UnitPrice * Quantity;
}
